### PR TITLE
Fixes #33736 - Allow 2 search with bookmark in one page

### PR DESCRIPTION
**Pull request by MariaAga**
The bookmark component uses `selectAutocompleteSearchQuery(state, 'searchBar', { controller }),` so only the search with id = 'searchBar' can be used with the bookmarks.

as I didn't find a page with 2 searches in foreman I tested it in foreman using:
`webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js` 
``` js
toolbarButtons={
        <div>
          <SearchBar
            {...{
              data: {
                controller: 'models',
                autocomplete: {
                  id: 'searchBar1',
                  searchQuery: '',
                  url: 'models/auto_complete_search',
                  useKeyShortcuts: true,
                },
                bookmarks: {
                  id: 'searchBar1',
                  url: '/api/bookmarks',
                  canCreate: true,
                  documentationUrl: '/links/manual/4.1.5Searching',
                },
              },
            }}
          />
          {canCreate && createBtn}
        </div>
      }
```

### CODE REVIEW

File: webpack/assets/javascripts/react_app/constants.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ export const getControllerSearchProps = (
     useKeyShortcuts: true,
   },
   bookmarks: {
+    id,
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/constants.js:22]: should it be called `autocompleteID` ?

File: webpack/assets/javascripts/react_app/components/PF4/Bookmarks/index.js [hunk 2]
```diff
@@ -28,15 +29,16 @@ const ConnectedBookmarks = ({
...
   );
   const isModalOpen = useSelector(store =>
-    selectIsModalOpen(store, BOOKMARKS_MODAL)
+    selectIsModalOpen(store, `${BOOKMARKS_MODAL}-${id}`)
   );
   const dispatch = useDispatch();
 
   const { setModalOpen, setModalClosed } = useForemanModal({
-    id: BOOKMARKS_MODAL,
+    id: `${BOOKMARKS_MODAL}-${id}`,
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/PF4/Bookmarks/index.js]: can you use a const for it somewhere above? 
e.g: 
```js
const modalID = `${BOOKMARKS_MODAL}-${id}`;
```

File: webpack/assets/javascripts/react_app/components/PF4/Bookmarks/index.js [hunk 3]
```diff
@@ -62,6 +65,7 @@ ConnectedBookmarks.propTypes = {
 };
 
 ConnectedBookmarks.defaultProps = {
+  id: '',
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/PF4/Bookmarks/index.js]: should the default `id` be the same as the search bar default id?

File: webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js [hunk 4]
```diff
@@ -7,20 +7,23 @@ import { noop } from '../../../common/helpers';
...
   onEnter,
   title,
   controller,
   url,
   bookmarks,
+  isPF4,
 }) => (
   <ForemanModal
-    id={BOOKMARKS_MODAL}
+    id={isPF4 ? `${BOOKMARKS_MODAL}-${id}` : BOOKMARKS_MODAL}
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: can you use the same helper to generate the PF4 modal id?

Also, is the PF3 bookmark used anywhere? thinking if we should drop it to simplify things.

File: webpack/assets/javascripts/react_app/constants.js (hunk 1 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/constants.js:22]: Not sure, it's also used for the id of the bookmarks modal

File: webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js (hunk 4 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: If it's not used we can remove it in a new pr

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: added  a helper

File: webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js [hunk 5]
```diff
@@ -5,22 +5,26 @@ import { BOOKMARKS_MODAL } from '../BookmarksConstants';
...
   onEnter,
   title,
   controller,
   url,
   bookmarks,
+  isPF4,
 }) => (
   <ForemanModal
-    id={BOOKMARKS_MODAL}
+    id={isPF4 ? getBookmarksModalId(id) : BOOKMARKS_MODAL}
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: since we count on the `id` in both bookmarks implementations, let's use just the `getBookmarksModalId(id)`
and drop the `isPF4` flag. It will also make it easier to remove the old bookmarks later on

File: webpack/assets/javascripts/react_app/routes/Models/constants.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ export const MODELS_PAGE_HIDE_LOADING = 'MODELS_PAGE_HIDE_LOADING';
 export const MODELS_PAGE_SHOW_LOADING = 'MODELS_PAGE_SHOW_LOADING';
 export const MODELS_PAGE_CLEAR_ERROR = 'MODELS_PAGE_CLEAR_ERROR';
 
-export const MODELS_SEARCH_PROPS = getControllerSearchProps('models');
+export const MODELS_SEARCH_PROPS = getControllerSearchProps('models','models');
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/routes/Models/constants.js]: is this change needed, or is it a leftover from when testing it on the models page?

**Review by Ron-Lavi — COMMENTED** [review]
Thanks @MariaAga ! tested and works well, just a few suggestions

File: webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js (hunk 5 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: Not sure what you mean? `webpack/assets/javascripts/react_app/components/Bookmarks/Bookmarks.js` doesn't pass `id` to the `SearchModal`

File: webpack/assets/javascripts/react_app/routes/Models/constants.js (hunk 6 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/routes/Models/constants.js]: Not needed, nice catch!

File: webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js (hunk 5 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: oh I see, so how about removing the `isPF4` flag and in `getBookmarksModalId` do something like this:
```js
export const getBookmarksModalId = id => id ? `${BOOKMARKS_MODAL}-${id}` : BOOKMARKS_MODAL;
  ```

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/Bookmarks/components/SearchModal.js]: Yeah, changed!

**Review by ezr-ondrej — APPROVED** [review]
Looks ready to me :+1:
